src/System.Linq/src/System/Linq/SelectMany.cs

```diff
@@ -133,5 +123,153 @@ public static partial class Enumerable
                 }
             }
         }
+
+        private sealed class SelectManySingleSelectorIterator<TSource, TResult> : Iterator<TResult>, IIListProvider<TResult>
```

Comment:
All of the changes here (except for `ToArray`) and the corresponding test changes are the result of pulling in #13942.